Foundation/JSONSerialization.swift

```diff
@@ -99,8 +104,11 @@ open class JSONSerialization : NSObject {
             // object is NSNumber and is not NaN or infinity
             // For better performance, this (most expensive) test should be last.
             if let number = _SwiftValue.store(obj) as? NSNumber {
-                let invalid = number.doubleValue.isInfinite || number.doubleValue.isNaN
-                return !invalid
+                if CFNumberIsFloatType(number._cfObject) {
+                    return number.doubleValue.isFinite
```

Comment:
This line made me do a double-take for a moment because at first glance, it's hard to visually distinguish between `number.doubleValue.isFinite` and `number.doubleValue.isInfinite`. Do you mind just calling this out with a comment saying something like "All floating-point numbers except for NaN and infinity are valid." (or similar)?

Reply:
Sure, I will actually put it back to how it was but store the `doubleValue` in a local since getting the `doubleValue` twice involved into 2 calls into `CFNumber` so was a significant speedup.